docs/source/basics.rst

```diff
@@ -176,6 +176,25 @@ There is more information about creating stubs in the
 The following sections explain the kinds of type annotations you can use
 in your programs and stub files.
 
+Mypy will also look in some default install directory to try to find annotations.
+For example, for UNIX based, with python 3.4, it will try to find it in
+- ``${HOME}/.local/shared/typehints/python3.4/``
+- ``/usr/shared/typehints/python3.4/``
```

Comment:
It is actually based on the example in PEP 484 in chapter [store and distribute stub files](https://www.python.org/dev/peps/pep-0484/#storing-and-distributing-stub-files) on how to add it to a setup.py.
```python
...
data_files=[
    (
        'shared/typehints/python{}.{}'.format(*sys.version_info[:2]),
        pathlib.Path(SRC_PATH).glob('**/*.pyi'),
    ),
],
...
```

> The standard directory is /usr/share/ (no d).

From the example, it is indeed 'shared', the '/usr' is because setup.py will install `data_files` under `sys.prefix` on system-wide install (based on [install info](https://docs.python.org/3/install/#alternate-installation-unix-the-prefix-scheme) on docs.python.org)

> And I think it would be better to put python* before typehints in the path, to mirror /usr/lib/pythonX.Y/.

That's taken from the example, so that would require a PEP change, but I guess it can easily be introduce as it isn't really implemented for now.

> I'd also rather use /usr/lib/python* for this rather than introduce a second python tree under /usr/share (and maybe put it under site-packages?).

/usr/lib/python* is for modules, not really for related data; IMO both options are valid, because typehints are closely related to code, but in itself, it isn't necessary for the code to run, thus not going into site-packages. /usr/ is also the standard place to store modules' related data.